### ISSUE TITLE
Failed to create codec for: {mime=video/avc, width=720, height=1280}

### ISSUE THREAD

**Issue by eightyfive** [apivideo/api.video-reactnative-live-stream#39]
**Describe the bug**
README example failing on Samsung Galaxy J7 Prime device (Android).

```
Error while updating property 'video' of a view managed by: ReactNativeLiveStreamView

null

Failed to create codec for: {mime=video/avc, width=720, height=1280}

[stack trace]
```

**To Reproduce**
1. Try to run the README example

**Expected behavior**
App to not crash. Maybe handle unsupported codec on devices gracefully?

**Smartphone (please complete the following information):**
 - Device: Samsung Galaxy J7 Prime
 - OS: Android
 - Version `1.2.1`

**Additional context**
I packaged `react-native-livestream` into custom Expo `dev-client`.

**Comment by eightyfive** [comment]
It works with `360p` resolution.

```ts
<LiveStreamView
        video={{
          fps: 30,
          resolution: "360p",
          bitrate: 2 * 1024 * 1024, // # 2 Mbps
        }}
```

How can I detect what resolution/codec is supported on my device?

**Comment by eightyfive** [comment]
Instead of opening a new issue, I am gonna ask here for now instead.

Still running the README example. I can make it work with `"resolution": "360p"` as explained above. But now, after I `stopStreaming` the first time, the "camera view" freezes.

This is how it goes:

1. Mount screen ✅ 
2. Do not start streaming yet
3. Move camera around, the "camera view" works ✅ 
4. Start streaming
5. `onConnecttionSuccess` event is fired ✅ 
6. Move camera around, the "camera view" works ✅ 
7. Stop streaming
8. `onDisconnect` event is fired ✅ 
9. Move camera around, the "camera view" works ✅ 
10. Start streaming again
11. `onConnecttionSuccess` event is fired ✅ 
12. "Camera view" _does not work_ ❌ 
13. Stop streaming
14. Button changes color red --> white (so not the _screen_ freezing) ✅ 
15. `onDisconnect` event is _NOT_ fired ❌ 

No JS errors, no crash. Just the "camera view" freezing.

Any help/hint would be appreciated 🙏

**Comment by eightyfive** [comment]
So if I re-mount the Screen. The "camera view" works again.

**Comment by eightyfive** [comment]
Follow-up:

I could get it to work by mounting/unmounting `<LiveStreamView />` based on `streaming: bool`.

While everything now "works", I'm not closing the issue just yet. If possible I would like some background about:

1. How to detect supported codec on device (if possible)?
2. Is the mounting/unmounting required or is it a bug?

Cheers 🙏

**Comment by ThibaultBee** [comment]
Nice.

1/ So, could you check that you got the same issue with that application with the same settings (h.264 and 720p)?  https://github.com/ThibaultBee/StreamPack/suites/8737375436/artifacts/395412454
2/ Check it does not happen with the same application (but with a small change) https://github.com/ThibaultBee/StreamPack/suites/8758666052/artifacts/396981234

**Comment by ThibaultBee** [comment]
I am surprised that your phone does not have an AVC encoder that support 720p.
Maybe an application that dump encoder and their capabilities would be nice.

**Comment by ThibaultBee** [comment]
Could you install a tool like that: https://play.google.com/store/apps/details?id=com.parseus.codecinfo&hl=en_US&gl=US
Find all AVC/h.264 video encoders, dump or screenshot their info and upload everything here?

**Comment by ThibaultBee** [comment]
I don't get yet why Android does not return this codec when we request an h.264 encoder that supports 720p.
Maybe because it is a software encoder?...

**Comment by eightyfive** [comment]
I just tried with another Android device and both issues are resolved.

Now the strange thing, they both run: **Android 8.1.0 Oreo** (API 26).

So it's really a hardware issue:

- KO: Galaxy J7 Prime
- OK: Razer Phone 2

**Comment by ThibaultBee** [comment]
If you look at the encoder in your second phone, I am pretty sure that it has an AVC/H.264 hardware encoder. That's why.

Could you test this APK: https://github.com/ThibaultBee/StreamPack/suites/8771926395/artifacts/397982074
I have forced the usage of "OMX.google.h264.encoder".

**Comment by ThibaultBee** [comment]
Hi,
This crash seems to be related to few models of Samsung. On these models, for an unknown reason, the `MediaCodec` API does not find an encoder that matches the expected configuration althought there is such encoder in the phone.
We are waiting to find one of this device. We will let you know when this issue is fixed.

**Comment by ThibaultBee** [comment]
Hi,

Could anyone test with https://github.com/ThibaultBee/StreamPack/suites/9154470311/artifacts/425992879?

**Comment by endrits079** [comment]
@ThibaultBee tested it with the device I was having trouble and seems to be working, btw do you have any library to play hls or rtmp?

**Comment by ThibaultBee** [comment]
Sorry, we haven't tested any RN player.

**Comment by ThibaultBee** [comment]
Hi, 

Could you check that your issue is fixed in the `main` branch?

**Comment by ThibaultBee** [comment]
This should be fixed in v1.2.2.
